python/kfserving/kfserving/kfmodel.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2019 kubeflow.org.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict
+
+import json
+import requests
+import tornado.web
+
+PREDICTOR_URL_FORMAT = "http://{0}/v1/models/{1}:predict"
+EXPLAINER_URL_FORMAT = "http://{0}/v1/models/{1}:explain"
+
+# KFModel is intended to be subclassed by various components within KFServing.
+class KFModel():
+
+    def __init__(self, name: str):
+        self.name = name
+        self.ready = False
+
+    def load(self):
+        self.ready = True
+
+    def preprocess(self, request: Dict):
+        return request
+
+    def postprocess(self, request: Dict):
+        return request
+
+    def predict(self, request: Dict) -> Dict:
+        if self.predictor_host is None:
```

Comment:
self.predictor_host seems not to be defined elsewhere. Is this a WIP?

Reply:
This comes from subclasses. I can set it to None in the init.